samples/user-interface/constraintlayout/src/main/java/com/example/platform/ui/constraintlayout/MotionLayout.kt

```diff
@@ -0,0 +1,235 @@
+    description = "Use a simple keyframe to change the interpolated motion",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class KeyframePositionFragment : Fragment(R.layout.motion_06_keyframe)
+
+@Sample(
+    name = "MotionLayout - 07. Keyframe",
+    description = "More complex keyframe, adding rotation interpolation",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class KeyframeInterpolationFragment : Fragment(R.layout.motion_07_keyframe)
+
+@Sample(
+    name = "MotionLayout - 08. Keyframe",
+    description = "Basic example of using a keyframe cycle",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class KeyframeCycleFragment : Fragment(R.layout.motion_08_cycle)
+
+@Sample(
+    name = "MotionLayout - 09. CoordinatorLayout 1",
+    description = "Basic example of using MotionLayout instead of AppBarLayout",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class Coordinator1Fragment : Fragment(R.layout.motion_09_coordinatorlayout)
+
+@Sample(
+    name = "MotionLayout - 10. CoordinatorLayout 2",
+    description = "Slightly more complex example of MotionLayout replacing AppBarLayout, with multiple elements and parallax background",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class Coordinator2Fragment : Fragment(R.layout.motion_10_coordinatorlayout)
+
+@Sample(
+    name = "MotionLayout - 11. CoordinatorLayout 3",
+    description = "Another AppBarLayout replacement example",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class Coordinator3Fragment : Fragment(R.layout.motion_11_coordinatorlayout)
+
+@Sample(
+    name = "MotionLayout - 12. DrawerLayout 1",
+    description = "Basic DrawerLayout with MotionLayout",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class Drawer1Fragment : Fragment(R.layout.motion_12_drawerlayout)
+
+@Sample(
+    name = "MotionLayout - 13. DrawerLayout 2",
+    description = "Advanced DrawerLayout with MotionLayout",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class Drawer2Fragment : Fragment(R.layout.motion_13_drawerlayout)
+
+@Sample(
+    name = "MotionLayout - 14. SidePanel",
+    description = "Side Panel, implemented with MotionLayout only",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class SidePanelFragment : Fragment(R.layout.motion_14_side_panel)
+
+@Sample(
+    name = "MotionLayout - 15. Parallax",
+    description = "Parallax background. Drag the car.",
+    documentation = "https://developer.android.com/develop/ui/views/animations/motionlayout",
+)
+class ParallaxFragment : Fragment(R.layout.motion_15_parallax)
+
+// TODO
```

Comment:
What's the problem of this one?

Reply:
Each of the remaining samples contains its own implementation. I thought that it's easier to review if I keep this PR restricted to simple XML samples. I can add the remaining ones in this PR if it's OK.

Reply:
Yes go ahead. I'm not checking the exact implementation/content since this is purely migration

Reply:
Done.